_test/test_sqw/test_PixelData.m

```diff
@@ -885,6 +885,155 @@ function test_unedited_dirty_pages_are_not_rewritten(obj)
+        pix_to_append = PixelData(appended_data);
+
+        pix.append(pix_to_append);
+
+        pix.move_to_first_page();
+        assertEqual(pix.data, existing_data, '', 1e-7);
+
+        pix.advance();
+        assertEqual(pix.data, appended_data, '', 1e-7);
+    end
+
+    function test_appending_pixels_after_page_edited_preserves_changes(obj)
+        npix_in_page = 11;
+        num_pix = 24;
+        mem_alloc = npix_in_page*obj.NUM_COLS_IN_PIX_BLOCK*obj.NUM_BYTES_IN_VALUE;
+        original_data = rand(9, num_pix);
+
+        pix = PixelData(original_data(:, 1:npix_in_page), mem_alloc);
+        assertEqual(pix.data, original_data(:, 1:npix_in_page));
+        pix.signal = ones(1, npix_in_page);
+
+        for i = 2:ceil(num_pix/npix_in_page)
+            start_idx = (i - 1)*npix_in_page + 1;
+            end_idx = min(start_idx + npix_in_page - 1, num_pix);
+            pix.append(PixelData(original_data(:, start_idx:end_idx)));
+            assertEqual(pix.data, original_data(:, start_idx:end_idx));
+        end
+
+        pix.move_to_first_page();
+        expected_pg_1_data = original_data(:, 1:npix_in_page);
+        expected_pg_1_data(8, :) = 1;
+        assertEqual(pix.data, expected_pg_1_data, '', 1e-7);
+    end
+
+    function test_you_can_append_to_file_backed_PixelData(obj)
+        npix_in_page = 11;
+        data = rand(9, 25);
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        data_to_append = rand(9, 8);
+        pix_to_append = PixelData(data_to_append);
+
+        pix.append(pix_to_append);
+
+        expected_final_pg = horzcat(data(:, 23:end), data_to_append);
+        assertEqual(pix.data, expected_final_pg);
+
+        pix.move_to_first_page();
+        assertEqual(pix.data, data(:, 1:npix_in_page), '', 1e-7);
+
+        pix.advance();
+        assertEqual(pix.data, data(:, (npix_in_page + 1):(2*npix_in_page)), ...
+                    '', 1e-7);
+
+        pix.advance();
+        assertEqual(pix.data, expected_final_pg, '', 1e-7);
+    end
+
+    function test_error_if_append_called_with_non_PixelData_object(~)
+        pix = PixelData(rand(9, 1));
+        f = @() pix.append(rand(9, 1));
+        assertExceptionThrown(f, 'PIXELDATA:append');
+    end
+
+    function test_error_if_appending_pix_with_multiple_pages(obj)
+        npix_in_page = 11;
+        mem_alloc = npix_in_page*obj.NUM_COLS_IN_PIX_BLOCK*obj.NUM_BYTES_IN_VALUE;
+        pix = PixelData(rand(9, 5), mem_alloc);
+
+        pix_to_append = obj.get_pix_with_fake_faccess(rand(9, 23), npix_in_page);
+        f = @() pix.append(pix_to_append);
+        assertExceptionThrown(f, 'PIXELDATA:append');
+    end
+
+    function test_append_does_not_edit_calling_instance_if_nargout_eq_1(obj)
+        data = rand(9, 30);
+        npix_in_page = 11;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+        pix_to_append = PixelData(rand(9, 5));
+
+        out_pix = pix.append(pix_to_append);
```

Comment:
there is no `assertEqual` to check the appended pixels *are* included in the new PixelData return value